src/schema.ts

```diff
@@ -247,19 +273,32 @@ function timeSummary(timeunit: TimeUnit, summary: Summary): Summary {
 /**
  * @return a new unique object based off of the old unique count and a binning scheme
  */
-function binUnique(bin, oldUnique) {
+function binUnique(bin, oldUnique, excludeInvalid: boolean) {
```

Comment:
I think you're running into the same problem with bin here (that your cached value is inconsistent?)

Instead, why don't you always add `NaN` and `null` as separate categories in the `newUnique`